src/main/java/ch/njol/skript/expressions/ExprHealReason.java

```diff
@@ -25,60 +25,69 @@
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.skript.log.ErrorQuality;
+import ch.njol.skript.registrations.EventValues;
 import ch.njol.util.Kleenean;
 
-
 @Name("Heal Reason")
-@Description("The <a href='./classes.html#healreason'>heal reason</a> of a heal event. Please click on the link for more information.")
-@Examples({"on heal:",
-	"\tif heal reason = satiated:",
-	"\t\tsend \"You ate enough food and gained health back!\" to player"})
+@Description("The <a href='./classes.html#healreason'>heal reason</a> of a <a href='./events.html#heal'>heal event</a>.")
+@Examples({
+	"on heal:",
+		"\theal reason is satiated",
+		"\tsend \"You ate enough food and gained full health back!\""
+})
+@Events("heal")
 @Since("2.5")
 public class ExprHealReason extends SimpleExpression<RegainReason> {
-	
+
 	static {
-		Skript.registerExpression(ExprHealReason.class, RegainReason.class, ExpressionType.SIMPLE, "(regen|health regain|heal) (reason|cause)");
+		Skript.registerExpression(ExprHealReason.class, RegainReason.class, ExpressionType.SIMPLE, "(regen|health regain|heal[ing]) (reason|cause)");
 	}
-	
+
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (!getParser().isCurrentEvent(EntityRegainHealthEvent.class)) {
-			Skript.error("Heal reason can only be used in an on heal event", ErrorQuality.SEMANTIC_ERROR);
+			Skript.error("The expression 'heal reason' may only be used in a healing event");
 			return false;
 		}
 		return true;
 	}
-	
+
 	@Nullable
 	@Override
-	protected RegainReason[] get(Event e) {
-		if (!(e instanceof EntityRegainHealthEvent))
+	protected RegainReason[] get(Event event) {
+		if (!(event instanceof EntityRegainHealthEvent))
 			return null;
+		return new RegainReason[]{((EntityRegainHealthEvent) event).getRegainReason()};
+	}
 
-		return new RegainReason[]{((EntityRegainHealthEvent) e).getRegainReason()};
+	@Override
+	public boolean setTime(int time) {
+		if (time == EventValues.TIME_FUTURE)
+			return false;
+		return super.setTime(time);
 	}
```

Comment:
Is it really necessary to not allow future reasons specifically? I don't really see the point. SimpleExpression already returns false for any setTime call.

Reply:
That means nothing. The default is that it doesn't have a time state, so it returns false. This allows for `if the heal reason was ...`